evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
```

Comment:
I'm 👍  with a1752cc .  I've futzed with this API enough that I've lost track of some of the "why" certain things are the way they are.  Instance level attribute makes sense and is easier to reason about.  Lets see if we run into any situations where this doesn't work.